addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/check-launch-dependency/check-launch-dependency.js

```diff
@@ -407,28 +424,67 @@ class CheckLaunchDependency extends StepBase {
   }
 
   async reportTimeout() {
-    const [stackName, resolvedVars] = await Promise.all([
+    const [stackName, resolvedVars, albLock] = await Promise.all([
       this.state.string('STACK_ID'),
       this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+      this.state.optionalString('ALB_LOCK'),
     ]);
     const envName = resolvedVars.name;
+    // Release ALB lock if exists
+    if (albLock) {
+      const [lockService] = await this.mustFindServices(['lockService']);
+      await lockService.releaseWriteLock({ writeToken: albLock });
+      this.print({
+        msg: `ALB lock released successfully`,
+      });
+    }
     throw new Error(
       `Error provisioning environment "${envName}". The workflow timed-out because the ALB provisioing stack "${stackName}" did not complete within the timeout period of 15 days.`,
     );
   }
 
+  async onPass() {
+    // this method is automatically invoked by the workflow engine when the step is completed
+    const [requestContext, resolvedVars, needsAlb, albLock] = await Promise.all([
+      this.payloadOrConfig.object(inPayloadKeys.requestContext),
+      this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+      this.payloadOrConfig.optionalBoolean(outPayloadKeys.needsAlb, false),
+      this.state.optionalString('ALB_LOCK'),
+    ]);
+    const [albService, lockService] = await this.mustFindServices(['albService', 'lockService']);
+    // Increase ALB dependent workspace count when there is a flag needs ALB
+    if (needsAlb) {
+      await albService.increaseAlbDependentWorkspaceCount(requestContext, resolvedVars.projectId);
+    }
+    // Release ALB lock if exists
+    if (albLock) {
+      await lockService.releaseWriteLock({ writeToken: albLock });
+      this.print({
+        msg: `ALB lock released successfully`,
+      });
+    }
+  }
```

Comment:
If albLock is `false`, isn't it an exception?

Reply:
Added additional checks to see if lock is present as per discussion